tests/integration_tests/test_upgrade.py

```diff
@@ -93,6 +96,26 @@ def test_upgrade(session_cloud: IntegrationCloud):
         instance.install_new_cloud_init(source, take_snapshot=False)
         instance.execute('hostname something-else')
         _restart(instance)
+        assert instance.execute('cloud-init status --wait --long').ok
         _output_to_compare(instance, after_path, netcfg_path)
 
     log.info('Wrote upgrade test logs to %s and %s', before_path, after_path)
+
+
+@pytest.mark.ci
+@pytest.mark.ubuntu
+def test_upgrade_package(session_cloud: IntegrationCloud):
+    if get_validated_source(session_cloud) != CloudInitSource.DEB_PACKAGE:
```

Comment:
Could this also work for -proposed, PPAs and `UPGRADE`? (Doesn't need to happen right now, ofc., but if we can expand the times we perform this test then why not?)

Reply:
IMO `test_upgrade` (a) does too many things, and (b) isn't really a test, but a way of using our testing framework to emit specialised output. As I mentioned above, introducing failure there is fine (and leaving this as-is is too, to be clear), but I do see this as a potential opportunity for us to decompose `test_upgrade` a little, into more specific tests.